packages/core/src/Button/index.tsx

```diff
@@ -497,3 +498,174 @@ export const IconButton = React.forwardRef<BaseElement, IconButtonProps>(
+              color: ${theme.color.text04()};
+              background-color: ${theme.color.element11(opacity[16])};
+            `
+          : undefined};
+      }
+      &:active {
+        box-shadow: 0 0 0 4px ${theme.color.elementPrimary(opacity[24])};
+        background-color: ${theme.color.element11(opacity[24])};
+      }
+      &:disabled {
+        opacity: ${opacity[48]};
+        cursor: default;
+        box-shadow: none;
+        &:hover {
+          ${IconContainer} {
+            background-color: ${theme.color.elementPrimary()};
+          }
+        }
+      }
+    `
+  }}
+`
+const IconContainer = styled(Icon)`
+  ${({ theme }) => {
+    return css`
+      height: ${componentSize.small};
+      width: ${componentSize.small};
+      color: ${theme.color.text00()};
+      background-color: ${theme.color.elementPrimary()};
+      border-radius: 2px;
+      margin-right: ${spacing.medium};
+      padding: ${spacing.small};
+      transition: all 200ms;
+    `
+  }}
+`
+interface IconTextButtonProps extends BaseProps {
+  /**
+   * String used to label the button.
+   */
+  readonly label: string
+  /**
+   * The icon element.
+   */
+  readonly icon: IconType
+
+  /**
+   * Specifies the name for a <button> element.
+   */
+  readonly name?: BaseProps['name']
+  /**
+   * `class` to be passed to the component.
+   */
+  readonly className?: BaseProps['className']
+  /**
+   * Specifies the type of button.
+   * Default: `button`
+   */
+  readonly type?: ButtonType
+}
+// eslint-disable-next-line react/display-name
+export const IconTextButton = React.forwardRef<
+  BaseElement,
+  IconTextButtonProps
+>(
+  (
+    {
+      disabled = false,
+      type = 'button',
+      icon,
+      onPointerDown,
+      onPointerUp,
+      onFocus,
+      label,
+      ...props
+    },
+    ref
+  ) => {
+    const {
+      isPointerOn,
+      isPointerOff,
+      determineVisibleFocus,
+      visibleFocus,
+    } = useVisibleFocus()
+
+    const handleFocus = useCallback<React.FocusEventHandler<BaseElement>>(
+      e => {
+        onFocus?.(e)
+        determineVisibleFocus()
+      },
+      [determineVisibleFocus, onFocus]
+    )
+    const handlePointerDown = useCallback<
+      React.PointerEventHandler<BaseElement>
+    >(
+      e => {
+        onPointerDown?.(e)
+        isPointerOn()
+      },
+      [isPointerOn, onPointerDown]
+    )
+    const handlePointerUp = useCallback<React.PointerEventHandler<BaseElement>>(
+      e => {
+        onPointerUp?.(e)
+        isPointerOff()
+      },
+      [isPointerOff, onPointerUp]
+    )
+    return (
+      <NativeIconTextButton
+        ref={ref}
+        disabled={disabled}
+        type={type}
+        icon={icon}
+        onPointerDown={handlePointerDown}
+        onPointerUp={handlePointerUp}
+        onFocus={handleFocus}
+        {...props}
+        visibleFocus={visibleFocus}
+      >
+        <Container>
+          {icon !== undefined ? <IconContainer icon={icon} /> : undefined}
```

Comment:
`icon` is required in props so the `icon` cannot be `undefined`

Reply:
Fixed.